accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -832,15 +673,15 @@ expressions. This means that we can allow those other non-redirecting
 generative constructors to coexist with a primary constructor.*
 
 A compile-time error occurs if the name of the primary constructor is the
-same as the name of a constructor (declaring or not) in the body.
+same as the name of a constructor in the body.
 
-*Moreover, it is an error if two constructor declarations in the body,
-declaring or otherwise, have the same name. This is just restating a
-compile-time error that we already have.*
+*Moreover, it is an error if two constructor declarations in the body have
+the same name. This is just restating a compile-time error that we already
+have.*
```

Comment:
And the primary constructor should also just be introducing a constructor with a name, and use the same rules as every other constructor declaration for avoiding conflicts.

Using special rules depending on the syntax being used is almost a guarantee to mis a case somewhere,